apps/sms/js/thread_list_ui.js

```diff
@@ -393,7 +393,8 @@ var ThreadListUI = {
     }
   },
 
-  markReadUnread: function thlui_markReadUnread(selected, isRead) {
+  markReadUnreadUI: function thlui_markReadUnreadUI(selected, isRead) {
+    var threadToMark = [];
     selected.forEach((id) => {
```

Comment:
if you use `filter` instead of `forEach`, you can make the code simpler. It takes a function: if the function returns "true", it keeps the element, otherwise it ignores it. Look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter for more information.